test/Atmos/EDMF/edmf_kernels.jl

```diff
@@ -377,12 +377,23 @@ function compute_gradient_flux!(
     en_dif.∇θ_liq = en_∇tf.θ_liq
     en_dif.∇q_tot = en_∇tf.q_tot
     en_dif.∇w = en_∇tf.w
-    # second moment env cov
+
     en_dif.∇tke = en_∇tf.tke
     en_dif.∇θ_liq_cv = en_∇tf.θ_liq_cv
     en_dif.∇q_tot_cv = en_∇tf.q_tot_cv
     en_dif.∇θ_liq_q_tot_cv = en_∇tf.θ_liq_q_tot_cv
 
+    # this fixes a BC problem with very large state⁻.ρatke (≈ e20)
+    # at the lower boundary:
+    z = altitude(m, aux)
+    if z==FT(0)
+        en_dif.∇tke = en_dif.∇tke*FT(0)
+        en_dif.∇θ_liq_cv = en_dif.∇θ_liq_cv*FT(0)
+        en_dif.∇q_tot_cv = en_dif.∇q_tot_cv*FT(0)
+        en_dif.∇θ_liq_q_tot_cv = en_dif.∇θ_liq_q_tot_cv*FT(0)
+    end
```

Comment:
This looks like a hack. We should find the problem instead.

Reply:
this is the point that I would like the help of the team in figuring out what is wrong. 
with out this hack state⁻.ρatke ≈ 1e20 which leads to a Qnorm=NaN. I am not sure why yet, maybe someone more versed with the structure of the BCs would have an idea. note that we are setting en⁺.ρatke in the BC function and its value is reasonable, O(0.1), when state⁻.ρatke  ≈ 1e20.